src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -20,6 +20,19 @@
 #include <config.h>
 #include "protocol.h"
 
+#define OLS_PACK_INT32(value, target) \
+	do { \
+		(target)[0] = (value) & 0xFF; \
+		(target)[1] = ((value) >> 8) & 0xFF; \
+		(target)[2] = ((value) >> 16) & 0xFF; \
+		(target)[3] = ((value) >> 24) & 0xFF; \
+	} while(0)
+
+#define OLS_PACK_INT16(value, target) \
+	do { \
+		(target)[0] = (value) & 0xFF; \
+		(target)[1] = ((value) >> 8) & 0xFF; \
+	} while(0)
```

Comment:
We already have WL16 and WL32 helpers in libsigrok-internal.h. Other than that, the 'send_longdata' function makes the code easier to read IMO.

Reply:
Thank you! I didn't know about this. Will use those instead.